drivers/cache/Kconfig

```diff
@@ -12,9 +12,10 @@ module = CACHE
 module-str = cache
 source "subsys/logging/Kconfig.template.log_config"
 
+endif # CACHE
+
```

Comment:
Why?

Reply:
I moved the CACHE_HAS_DRIVER because I thought it was needed, and the fact that it was under a default n menuconfig made it throw build errors. It might have been a mistake to move, I could have just selected the other CACHE config as well.

But I am also wondering now what is the purpose of the CACHE_HAS_DRIVER symbol, because I realized I don't even need to select it since it isn't used anywhere.

Also the name of the CACHE config seems too generic to me if it is just for describing the use of external caches.

Reply:
can you please add your comments / concerns to https://github.com/zephyrproject-rtos/zephyr/issues/50747 ? I'm trying to gather as much inputs as possible to rework the whole API in a way that makes sense.

Reply:
Following what other driver classes are doing this should be indeed under the `if CACHE`